### PR TITLE
Added the possibility to influence the exception handling behavior of the Dispatcher

### PR DESCRIPTION
**Pull request by dennisdoomen**
The `Dispatcher` class now has a property `ExceptionHandler` that can be used to change the default behavior of logging and aborting the dispatching process for a specific subscriber.

### CODE REVIEW

File: Src/LiquidProjections/Dispatcher.cs [hunk 1]
```diff
@@ -36,44 +39,98 @@ public Dispatcher(CreateSubscription createSubscription)
...
 
+    /// <summary>
+    /// Defines the signature for a method that handles an exception while dispatching transactions.
+    /// </summary>
+    /// <param name="exception">The exception that was caught by the <see cref="Dispatcher"/></param>
+    /// <param name="info">Information about the subscription.</param>
+    /// <returns>
+    /// Instructs the <see cref="Dispatcher"/> on how to resolve the exception.
+    /// </returns>
+    public delegate Task<ExceptionResolution> HandleException(Exception exception, SubscriptionInfo info);
```
Comment (IharBury) [inline Src/LiquidProjections/Dispatcher.cs]: Should we maybe allow the handler to know some context about retrying progress? Like something allowing the handler to retry the same commit only 3 times, for example.

File: Src/LiquidProjections/Dispatcher.cs [hunk 2]
```diff
@@ -36,44 +39,98 @@ public Dispatcher(CreateSubscription createSubscription)
...
+                            abort(exception);
+                            retry = false;
+                            break;
+
+                        case ExceptionResolution.Retry:
+                            break;
+
+                        case ExceptionResolution.Ignore:
+                            retry = false;
+                            ignore?.Invoke();
```
Comment (IharBury) [inline Src/LiquidProjections/Dispatcher.cs]: Should we catch and log exceptions thrown from the `abort` and `ignore` delegates?

Comment (dennisdoomen) [inline Src/LiquidProjections/Dispatcher.cs]: I think that's up to the caller to decide. I don't want to make decisions for them.

File: Src/LiquidProjections/Dispatcher.cs (hunk 1 above)
Comment (dennisdoomen) [inline Src/LiquidProjections/Dispatcher.cs]: I like that.

Comment (dennisdoomen) [inline Src/LiquidProjections/Dispatcher.cs]: Added this.

**Pull request merged**